### PR TITLE
feat: create List component

### PR DESCRIPTION
**Pull request by paulgrym**
## Jira issue

JIRA issue: [FSU-50](https://klau.atlassian.net/browse/FSU-50)

## Describe your changes

Created List component.

## Checklist before requesting a review

- [x] Checked which branch you would merge.
- [x] Checked if the code follows the style guidelines of this project (use eslint and prettier).
- [x] Performed a self-review of code (bugs, clean code rules, etc).
- [x] Checked if added variables have unique names.
- [x] Checked if added code/function is not a duplicate. 
- [x] Tests have been added if possible.
- [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: packages/design-system/components/list/List.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { node, string } from "prop-types"
+
+import * as Styled from "./List.styled"
+
+const List = ({ gap, children }) => (
+  <Styled.Component gap={gap}>{children}</Styled.Component>
+)
+
+List.propTypes = {
+  gap: string.isRequired,
```
Comment (im-klau) [inline packages/design-system/components/list/List.jsx:10]: I think we're going to use only `px` values which means we can use `number` as a prop type. What do you think?
```
<List gap={20} />

Comment (im-klau) [reply packages/design-system/components/list/List.jsx:10]: I'd say also, the `gap` property should have a default value

File: packages/design-system/components/list/__stories__/List.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# List
+
+<Canvas>
+  <Story
+    name="Overview"
+    args={{
+      gap: "8px",
+      children: (
+        <>
```
[the remark this answers is not in the document]
Comment (paulgrym) [reply packages/design-system/components/list/__stories__/List.stories.mdx]: Good idea, thanks :)

File: packages/design-system/components/list/List.styled.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import styled from "@emotion/styled"
+
+export const Component = styled.ul`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: ${({ gap }) => `${gap}px`};
+  list-style-type: none;
+  padding-left: 0;
+  margin: 0; //to change?
```
Comment (im-klau) [inline packages/design-system/components/list/List.styled.js:10]: I think the margin should be set to 0 to reset default styles:)

File: packages/design-system/components/list/__tests__/List.test.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import List from "../List"
+
+const children = <li>Test1</li>
+
+describe(`List`, () => {
+  it(`renders with default properties`, () => {
+    render(<List>{children}</List>)
+  })
+
+  it(`renders multiple list items when passed`, () => {
```
Comment (im-klau) [inline packages/design-system/components/list/__tests__/List.test.jsx:12]: wow, well done! 🚀

**Review by im-klau — APPROVED** [review]

File: packages/design-system/components/list/__stories__/List.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+<ArgsTable />
+
+export const Template = (args) => (
+  <List {...args}>
+    <ListItem iconName="done">Lista element 1 </ListItem>
+    <ListItem iconName="done" variant="gray">
+      Lista element 2
+    </ListItem>
+    <ListItem iconName="done">Lista element 3 </ListItem>
+  </List>
```
Comment (nata-kab) [inline packages/design-system/components/list/__stories__/List.stories.mdx:30]: Well done 🚀 Just one little thing - I can see extra space after "Lista element 1" and "Lista element 3". Please check it.🕵️‍♀️

**Review by nata-kab — APPROVED** [review]
I love how you done this task.👏 The code is clean and easy to understand. Great job 🎉

**Pull request merged**